util/stringz/utils.go

```diff
@@ -33,6 +33,16 @@ func Contains(a []string, val string) bool {
 	return found
 }
 
+func Remove(a []string, val string) []string {
```

Comment:
I can change it if you feel strongly about it, but from my perspective Remove implies RemoveAll and if we want RemoveFirst/RemoveLast/RemoveN those could sit next to Remove. 

Also, I'm hopeful that this will be a temporary thing until we get generics and this is added to the base library :)